src/metabase/models/serialization.clj

```diff
@@ -358,24 +363,27 @@
   - Replace any foreign keys with portable values (eg. entity IDs, or a user ID with their email, etc.)"
   [model-name opts instance]
   (try
-    (binding [*current* instance]
-      (let [spec (make-spec model-name opts)]
-        (assert spec (str "No serialization spec defined for model " model-name))
-        (-> (select-keys instance (:copy spec))
-            ;; won't assoc if `generate-path` returned `nil`
-            (m/assoc-some :serdes/meta (generate-path model-name instance))
-            (into (for [[k transform] (:transform spec)
-                        :let  [export-k (:as transform k)
-                               res ((:export transform) (get instance k))]
-                        :when (not= res ::skip)]
-                    (do
-                      (when-not (contains? instance k)
-                        (throw (ex-info (format "Key %s not found, make sure it was hydrated" k)
-                                        {:model    model-name
-                                         :key      k
-                                         :instance instance})))
-
-                      [export-k res]))))))
+    (let [spec (make-spec model-name opts)]
+      (assert spec (str "No serialization spec defined for model " model-name))
+      (-> (select-keys instance (:copy spec))
+          ;; won't assoc if `generate-path` returned `nil`
+          (m/assoc-some :serdes/meta (generate-path model-name instance))
+          (into (for [[k transform] (:transform spec)
+                      :let [_         (assert-one-defined transform :export :export-with-context)
+                            export-k  (:as transform k)
+                            input     (get instance k)
+                            f         (:export transform)
+                            f-context (:export-with-context transform)
+                            res       (if f (f input) (f-context instance k input))]
```

Comment:
The `[m k x]` ordering is inspired by the signature of atom watchers and just feels natural to me.

We discussed also always taking `x` as the first argument, so its as if the 1-arity functions just implicitly discard the rest of the arguments, like in Javascript. To me it "feels bad man" to put put "inner" objects before their parent in a signature. Basically I'm used to starting with the "widest" or "longest lived" arguments, its a bit tricky to articulate it, but tends to play well with currying.

Reply:
Maybe this is indeed some JS in me talking 😁